src/gui/mzroll/downloadmanager.h

```diff
@@ -0,0 +1,44 @@
+#ifndef DOWNLOADMANAGER_H
+#define DOWNLOADMANAGER_H
+#include <QObject>
+#include <QUrl>
+#include <QNetworkReply>
+
+class QNetworkAccessManager;
+class QNetworkRequest;
+class QUrl;
+
+class DownloadManager : public QObject
+{
+    Q_OBJECT
+public:
+    DownloadManager();
+    void setRequest(const QString& url, void* requester, bool async=true);
+    bool err;
+//    enum class RequestType {
+//        ASYNC,
+//        SYNC
+//    };
+
+
+private:
+    void download(bool async=true);
+
+public slots:
+    void dataAvailable();
+    void finished();
+    void error(QNetworkReply::NetworkError error);
```

Comment:
I think the above three slots can be either private or protected. No need to expose these, since only the manager needs to call to them.

Reply:
They are connected to QNetworkReply's signals, hence `public`

Reply:
I can only see these connections being made within this class itself. So making them private should not violate any rules, right?

Reply:
They will be called by `QNetworkReply` and not by `DownloadManager`.
Let me know if you still have doubts regarding this

Reply:
I am probably not understanding this well, but I made the slots private locally and it worked fine for me. That was all I was asking for.

Reply:
Thanks for checking, i just checked on my local as well and it worked. 
Wasn't expecting it to work though, I always thought they followed the same conventions as normal member functions but no, just read this on [Qt docs](https://doc.qt.io/qt-5/signalsandslots.html#slots):  "This means that a signal emitted from an instance of an arbitrary class can cause a private slot to be invoked in an instance of an unrelated class." 

Sorry my bad, will make this change